### PR TITLE
FEATURE: Use async search for category dropdown components

### PR DESCRIPTION
**Pull request by nbianca**
This makes it work better on sites with many categories.

### CODE REVIEW

File: app/controllers/categories_controller.rb [hunk 1]
```diff
@@ -297,6 +299,69 @@ def visible_groups
...
+    include_subcategories =
+      if params[:include_subcategories].present?
+        ActiveModel::Type::Boolean.new.cast(params[:include_subcategories])
+      else
+        true
+      end
+    prioritized_category_id = params[:prioritized_category_id].to_i if params[
+      :prioritized_category_id
+    ].present?
+    limit = params[:limit].to_i.clamp(1, MAX_CATEGORIES_LIMIT) if params[:limit].present?
```
Comment (ZogStriP) [inline app/controllers/categories_controller.rb:324]: Nice 👌

File: app/assets/javascripts/discourse/app/helpers/category-link.js [hunk 2]
```diff
@@ -185,11 +185,5 @@ export function defaultCategoryLinkRenderer(category, opts) {
   if (opts.topicCount && categoryStyle === "box") {
     afterBadgeWrapper += buildTopicCount(opts.topicCount);
   }
-  if (opts.plusSubcategories && opts.lastSubcategory) {
```
Comment (ZogStriP) [inline app/assets/javascripts/discourse/app/helpers/category-link.js:188]: Why was this deleted?

**Review by ZogStriP — APPROVED** [review]

File: app/assets/javascripts/discourse/app/helpers/category-link.js (hunk 2 above)
Comment (nbianca) [reply app/assets/javascripts/discourse/app/helpers/category-link.js:188]: It was such an obscure feature I doubt many people used. Then, it was hard to implement with the new async search model. I can add it back in the future if people want it back.
